apps/prairielearn/src/lib/editors.ts

```diff
@@ -906,18 +907,22 @@ export class AssessmentAddEditor extends Editor {
 
 export class CourseInstanceCopyEditor extends Editor {
   private course_instance: CourseInstance;
+  private from_course: Course;
+  private from_path: string;
 
   public readonly uuid: string;
 
-  constructor(params: BaseEditorOptions<{ course_instance: CourseInstance }>) {
-    const { course_instance } = params.locals;
-
-    super({
-      ...params,
-      description: `Copy course instance ${course_instance.short_name}`,
-    });
-
-    this.course_instance = course_instance;
+  constructor(
+    params: BaseEditorOptions & {
+      from_course: Course;
+      from_path: string;
```

Comment:
Checking my understanding: the reason we need a separate path instead of just constructing it from `from_course.path` and `course_instance.short_name` is because of the weird little file transfer thing? In other words, we're not actually copying things out of the source course directly?

I forget, did we ever figure out _why_things were done like this? If so, it'd be nice to document that somewhere.